src/Raven.Server/Https/ExternalCertificateValidator.cs

```diff
@@ -0,0 +1,262 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Net.Security;
+using System.Security.Cryptography.X509Certificates;
+using System.Text.RegularExpressions;
+using System.Threading;
+using System.Threading.Tasks;
+using Raven.Client.Http;
+using Sparrow;
+using Sparrow.Logging;
+using Sparrow.Utils;
+
+namespace Raven.Server.Https
+{
+    public class ExternalCertificateValidator
+    {
+        private readonly RavenServer _server;
+        private readonly Logger _logger;
+
+        private ConcurrentDictionary<Key, Task<CachedValue>> _externalCertificateValidationCallbackCache;
+
+        public ExternalCertificateValidator(RavenServer server, Logger logger)
+        {
+            _server = server;
+            _logger = logger;
+        }
+
+        public void Initialize()
+        {
+            if (string.IsNullOrEmpty(_server.Configuration.Security.CertificateValidationExec))
+                return;
+            
+            _externalCertificateValidationCallbackCache = new ConcurrentDictionary<Key, Task<CachedValue>>();
+
+            RequestExecutor.RemoteCertificateValidationCallback += (sender, cert, chain, errors) => ExternalCertificateValidationCallback(sender, cert, chain, errors, _logger);
+        }
+
+        public void ClearCache()
+        {
+            _externalCertificateValidationCallbackCache = new ConcurrentDictionary<Key, Task<CachedValue>>();
```

Comment:
`_externalCertificateValidationCallbackCache?.Clear()` wouldn't be better?

Reply:
yes